Zend/tests/float_to_int/warnings_float_literals.phpt

```diff
@@ -0,0 +1,147 @@
+--TEST--
+Implicit float to int conversions should warn for literals
+--FILE--
+<?php
+// TODO Assignment version and when float is a variable instead of literal
+// TODO Float strings for bitwise, arithmetics ops, and function calls/returns
+
+echo 'Bitwise ops:' . \PHP_EOL;
+// 8 Warnings generated in total
+$var = ~1.5;
+var_dump($var);
+$var = 1.5|3;
+var_dump($var);
+$var = 1.5&3;
+var_dump($var);
+$var = 1.5^3;
+var_dump($var);
+$var = 1.5 << 3;
+var_dump($var);
+$var = 1.5 >> 3;
+var_dump($var);
+$var = 3 << 1.5;
+var_dump($var);
+$var = 3 >> 1.5;
+var_dump($var);
+
+echo 'Modulo:' . \PHP_EOL;
+// 2 warnings in total
+$var = 6.5 % 2;
+var_dump($var);
+$var = 9 % 2.5;
+var_dump($var);
+
+// Exponentiation?
+
+echo 'Offset access:' . \PHP_EOL;
+echo 'Arrays:' . \PHP_EOL;
+// 2 warnings in total
+$array = ['a', 'b', 'c'];
+var_dump($array[1.5]);
+$array[2.5] = 'z';
+var_dump($array);
+
+echo 'Strings:' . \PHP_EOL;
+// 2 warnings in total
+$string = 'php';
+var_dump($string[1.5]);
+$string[2.5] = 'z';
+var_dump($string);
+
+echo 'Function calls:' . \PHP_EOL;
+function foo(int $a) {
+    return $a;
+}
+var_dump(foo(1.5));
+
+// TODO: Why are two warnings generated here?
+var_dump(chr(60.5));
```

Comment:
Still don't understand why calls to internal functions trigger 2 diagnostics to be produced.